src/en/rinkocomics/src/eu/kanade/tachiyomi/extension/en/rinkocomics/RinkoComics.kt

```diff
@@ -0,0 +1,231 @@
+
+            // [CRITICAL] Headers that make WordPress accept the request
+            val ajaxHeaders = headers.newBuilder()
+                .add("X-Requested-With", "XMLHttpRequest")
+                .add("Referer", request.url.toString())
+                .add("Origin", baseUrl)
+                .build()
+
+            if (comicId != null) {
+                while (true) {
+                    val formBody = FormBody.Builder()
+                        .add("action", "load_more_chapters")
+                        .add("comic_id", comicId)
+                        .add("offset", offset.toString())
+                        .build()
+
+                    // Request to admin-ajax.php
+                    val ajaxRequest = POST("$baseUrl/wp-admin/admin-ajax.php", ajaxHeaders, formBody)
+                    
+                    try {
+                        val ajaxResponse = client.newCall(ajaxRequest).execute()
+                        val ajaxHtml = ajaxResponse.body.string()
+
+                        // Stop if empty or explicitly says "no more"
+                        if (ajaxHtml.isBlank() || ajaxHtml.contains("no more chapters", true)) {
+                            break
+                        }
+
+                        // Parse the response (it returns raw <li> elements)
+                        val ajaxDoc = Jsoup.parseBodyFragment(ajaxHtml)
+                        val newChapters = ajaxDoc.select("li.chapter:not(.locked-chapter)")
+                            .map { chapterFromElement(it) }
+                            .filter { visitedUrls.add(it.url) }
+
+                        if (newChapters.isEmpty()) break
+
+                        chapters.addAll(newChapters)
+                        offset += 10
+
+                    } catch (e: Exception) {
+                        break
+                    }
+                }
+            }
+
+            chapters
+        }
+    }
+
+    // Exclude locked chapters from the selector
+    override fun chapterListSelector() = "ul.chapters-list li.chapter:not(.locked-chapter)"
+
+    override fun chapterFromElement(element: Element) = SChapter.create().apply {
+        val anchor = element.selectFirst("a")!!
+        setUrlWithoutDomain(anchor.attr("href"))
+        
+        name = element.selectFirst("span.chapter-number")?.text()?.trim() ?: anchor.text()
+        
+        val dateText = element.selectFirst("span.chapter-date")?.text()?.trim() ?: ""
+        date_upload = parseDate(dateText)
+    }
+
+    // =========================================================================
+    //  Page List
+    // =========================================================================
+
+    override fun pageListParse(document: Document): List<Page> {
+        val images = document.select("div.images-flow img.chapter-image")
+
+        return images.mapIndexed { index, img ->
+            val url = (img.attr("abs:data-src").ifBlank { img.attr("abs:src") }).trim()
```

Comment:
```suggestion
            val url = (img.attr("abs:data-src")).trim()
```
`.ifBlank` shouldn't be necessary. Not sure if `.trim()` is required -- Jsoup docs doesn't mention if this is pre-trimmed for you.